api/server/router/image/image_routes.go

```diff
@@ -432,9 +432,17 @@ func (ir *imageRouter) getImagesJSON(ctx context.Context, w http.ResponseWriter,
 		return err
 	}
 
+	manifests := httputils.BoolValue(r, "manifests")
+	if versions.LessThan(version, "1.47") {
+		manifests = false
+	}
+
 	useNone := versions.LessThan(version, "1.43")
 	withVirtualSize := versions.LessThan(version, "1.44")
 	for _, img := range images {
+		if !manifests {
+			img.Manifests = nil
+		}
```

Comment:
Hmm, I considered that initially but dropped the idea because we need to traverse the manifests anyway.

However, we could still avoid the allocation host. So might be worth to do it this way.